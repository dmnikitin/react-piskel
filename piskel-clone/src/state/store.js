import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { getFromLocalStorage } from '../helpers/localStorageHandler';
import combined from './reducers/combined';

const localStorageData = getFromLocalStorage();
const store = createStore(combined, localStorageData, composeWithDevTools());

export default store;
