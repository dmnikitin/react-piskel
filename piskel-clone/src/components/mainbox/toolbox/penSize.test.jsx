import React from 'react'
import { shallow, mount } from 'enzyme';
import PenSize from './penSize';
import store from '../../../state/store';

const callback = () => { };
jest.mock('../button.jsx', () => 'Button');
const wrapperShallow = shallow(<PenSize callback={callback} />);
const wrapperMount = mount(<PenSize store={store} callback={callback} />);

describe('pensize component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it("displays initial div", () => {
    expect(wrapperMount.find("div")).toHaveLength(2);
  });
  it('displays proper amount of buttons', () => {
    expect(wrapperMount.find('.toolbox-pensize__buttons').children()).toHaveLength(3);
  });
  it('displays heading', () => {
    expect(wrapperMount.find('h3').text()).toBe('Choose the pen size');
  });
});
