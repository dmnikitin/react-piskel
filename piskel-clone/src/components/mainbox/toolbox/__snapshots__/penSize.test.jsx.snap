// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pensize component renders correctly 1`] = `
<div
  className="toolbox-pensize"
>
  <h3>
    Choose the pen size
  </h3>
  <div
    className="toolbox-pensize__buttons"
  >
    <Button
      callback={[Function]}
      data="small"
      icon="pencil"
    />
    <Button
      callback={[Function]}
      data="medium"
      icon="pencil-square-o"
    />
    <Button
      callback={[Function]}
      data="large"
      icon="pencil-square"
    />
  </div>
</div>
`;
